apps/passwd.c

```diff
@@ -47,6 +33,14 @@ static unsigned const char cov_2char[64] = {
     0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A
 };
 
+static const char ascii_dollar[] = { 0x24, 0x00 };
+#if defined(CHARSET_EBCDIC) && !defined(CHARSET_EBCDIC_TEST)
+static const char ebcdic_dollar[] = { 0x5B, 0x00 };
+# define dollar ebcdic_dollar
+#else
+# define dollar ascii_dollar
+#endif
```

Comment:
As for `dollar[] = "$"`. Note that it works without any #ifdefs on any case. With -DCHARSET_EBCDIC, with -DCHARSET_EBCDIC and -DCHARSET_EBCDIC_TEST, without -DCHARSET_EBCDIC on ASCII platform...

As for ctype replacement. In addition to what was said, I'd add that prime goal was to *ensure* independence on locale, on ASCII platforms[!], when explicitly appropriate. Concern is that we have no control over locale target system/application/user uses.